src/server/qgsserverlogger.cpp

```diff
@@ -36,44 +36,42 @@ QgsServerLogger *QgsServerLogger::instance()
 }
 
 QgsServerLogger::QgsServerLogger()
-  : mLogFile( nullptr )
+  : QgsMessageLogConsole()
 {
-  connect( QgsApplication::messageLog(), static_cast<void ( QgsMessageLog::* )( const QString &, const QString &, Qgis::MessageLevel )>( &QgsMessageLog::messageReceived ), this,
-           &QgsServerLogger::logMessage );
 }
 
-void QgsServerLogger::setLogLevel( Qgis::MessageLevel level )
+void QgsServerLogger::logMessage( const QString &message, const QString &tag, Qgis::MessageLevel level )
+{
+  if ( mLogLevel > level )
+  {
+    return;
+  }
+  if ( mLogFile.isOpen() )
+  {
+    QString formattedMessage = formatLogMessage( message, tag, level );
+    mTextStream << formattedMessage;
+    mTextStream.flush();
+  }
+  else if ( mLogFile.fileName() == "stderr" )
```

Comment:
What do you think about using `QString::compare()` with the `Qt::CaseInsensitive` flag?

Reply:
Well I don't know. "stderr" being a special value, I would be tempted to just support "stderr". But I don't care too much. If you think it's better to support all the combinations that's ok to me too. You decide :)

Reply:
Actually, I'm not sure of that either... But, for example, I think that 'STDERR' is used sometimes on Windows. 

So I don't have a strong opinion, but I think it would be a more robust approach :).

Reply:
I decided that you were right and went for the more robust approach :)